staging/src/k8s.io/csi-translation-lib/translate.go

```diff
@@ -67,7 +67,20 @@ func (CSITranslator) TranslateInTreeInlineVolumeToCSI(volume *v1.Volume) (*v1.Pe
 	}
 	for _, curPlugin := range inTreePlugins {
 		if curPlugin.CanSupportInline(volume) {
-			return curPlugin.TranslateInTreeInlineVolumeToCSI(volume)
+			pv, err := curPlugin.TranslateInTreeInlineVolumeToCSI(volume)
+			if err != nil {
+				return nil, err
+			}
+			// Inline volumes only support PersistentVolumeFilesystem (and not block).
```

Comment:
Does this cause issues in its current state? Could you pull this change out into it's own PR and link it here - seems like an unrelated bug-fix that we could probably get in earlier

Reply:
Very good point - this specific change used to be necessary due to changes introduced in this PR. The kubelet DSOW populator [used](https://github.com/kubernetes/kubernetes/pull/84048/files#diff-94e527d26427ceb4f990f05f76952656L510) to invoke `GetVolumeMode` on a volume and this failed when migration is enabled. However [this recent change](https://github.com/kubernetes/kubernetes/pull/84048/files#diff-94e527d26427ceb4f990f05f76952656L510) relaxed this a bit but I kept this around for now to be safe (so that things don't break when the [TODO](https://github.com/kubernetes/kubernetes/pull/84048/files#diff-94e527d26427ceb4f990f05f76952656R508) is addressed and migration tests are not immediately run). I will run a test without the above change and if tests still pass for inline with migration enabled, I will separate it out from this PR.

Reply:
We will still need the change to handle [CheckVolumeFileSystem](https://github.com/kubernetes/kubernetes/blob/14e5adfc85119b0fca6bed1dabfca8bc94d9f923/pkg/volume/util/operationexecutor/operation_executor.go#L757) calls which call into `GetVolumeMode`